yt_dlp/extractor/facebook.py

```diff
@@ -583,8 +612,8 @@ def extract_relay_data(_filter):
         def extract_relay_prefetched_data(_filter):
             return traverse_obj(extract_relay_data(_filter), (
                 'require', (None, (..., ..., ..., '__bbox', 'require')),
-                lambda _, v: 'RelayPrefetchedStreamCache' in v, ..., ...,
-                '__bbox', 'result', 'data', {dict}), get_all=False) or {}
+                lambda _, v: any(key.startswith('RelayPrefetchedStreamCache') for key in v),
+                ..., ..., '__bbox', 'result', 'data', {dict}), get_all=False) or {}
```

Comment:
IMO, this change seems unrelated to any of the auth error handling and should be merged separately, as this is what actually partially addresses #4311, correct?

Reply:
#4311 is fairly unspecific on it's own because the error handling is led in this way. Hence why different things pop up as a result for this. This is one of them. I can separate it but it would still be addressing the same issue.

Reply:
Would you like me to extract it to another PR?

Reply:
committed to master and merged into this PR